### ISSUE TITLE
[BUG] etcd restore on an RKE1 cluster goes into `Updating` state with error `Backing off failure, delay: 30s` but the etcd restore finishes

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#40889]
**Rancher Server Setup**
- Rancher version: 2.7-head commit id: `cd1a716 `
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2 - `v1.24.10+rke2r1`

**Information about the Cluster**
- Kubernetes version: `1.24.10` to `v1.25.6`
- Cluster Type (Local/Downstream): Downstream - 3 etcd, 2 cp and 3 worker nodes

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) admin

**Describe the bug**
[BUG] etcd restore on an RKE1 cluster goes into `Updating` state with error `Backing off failure, delay: 30s` but the etcd restore finishes

**To Reproduce**

- Deploy an RKE1 1.24 cluster with PSP enabled - set to Unrestricted
- Take snapshot b1
- Edit cluster upgrade k8s to 1.25. Disable PSP and save changes made
- Cluster upgrades successfully
- Restore to etcd snapshot b1 taken in step 2
- cluster goes into `Updating` state with error `Backing off failure, delay: 30s` but the etcd restore finishes

**Expected Result**
Error should NOT be seen on the cluster when snapshot is in progress

---

**Note:**
- This is not seen on an upgrade from 1.23 to 1.24 and restore to 1.23 with psp enabled
- This is also not seen on an upgrade from 1.24 to 1.25 and restore to b1 on 1.24 with no psp enabled initially.

**Comment by jiaqiluo** [comment]
When Rancher restores the etcd snapshot,  one of the steps is to update the cluster ( [code](https://github.com/rancher/rancher/blob/f2faf1a8aa1081301e1fe6c21637fe9b7044f188/pkg/controllers/management/clusterprovisioner/driver.go#L136-L139)), which fails because of conflicting (see error msg below)  if PSP is disabled in the cluster's current spec but was enabled in the backup (and vice versa). 

This failure is recorded by the reconciling handler and triggers the backoff of the reconciling process. ([code](https://github.com/rancher/rancher/blob/f2faf1a8aa1081301e1fe6c21637fe9b7044f188/pkg/controllers/management/clusterprovisioner/provisioner.go#L527-L529)) But this does not mean the restoring fails: it will keep processing, and the cluster becomes active once it is done. 

TLDR, the restoring function works fine; it is the state and error message in the UI during the process looks "scary" although UI is reflecting the right information it reads from the cluster object . 

The conflicting error message: 
```
Operation cannot be fulfilled on clusters.management.cattle.io "c-gdzsq": the object has been modified; please apply your changes to the latest version and try again
```


Solutions:
The best fix is to fix the conflicts on the cluster update. Honestly I have not figured out what causes the conflict. 
Also, We can improve the error message from the bacoff function to make it sure that the backoff happens because restore is process.

**Comment by jiaqiluo** [comment]
@kinarashah and I discussed this issue offline, we found the call to to update the cluster might be unnecessary because the driverUpdate function and its caller do not modify the cluster. Also, only the [driverUpdate](https://github.com/rancher/rancher/blob/f2faf1a8aa1081301e1fe6c21637fe9b7044f188/pkg/controllers/management/clusterprovisioner/driver.go#L136-L139) function to return the error, but  `driverRemove` and `driverUpdate` and `driverCreate`  ignore the error.


The same issue is reproduced in rancher v2.6.9. 
Because this is not a regression, we would like to move it out of the 2.7.2 milstone. 
![screencapture-164-92-64-23-dashboard-c-manager-provisioning-cattle-io-cluster-2023-03-17-14_02_26](https://user-images.githubusercontent.com/6218999/226048799-dd19c872-8f26-4f43-bb36-6175d3f6e073.png)

**Comment by jiaqiluo** [comment]
### Root cause

TLDR:  An unnecessary return of error on the update on the cluster object causes conflicting. 

When Rancher restores the etcd snapshot, one of the steps is to [update the cluster](https://github.com/rancher/rancher/blob/f2faf1a8aa1081301e1fe6c21637fe9b7044f188/pkg/controllers/management/clusterprovisioner/driver.go#L136-L139), which fails because of conflicting (see error msg below) if PSP is disabled in the cluster's current spec but was enabled in the backup (and vice versa).

The conflicting error message:
```
Operation cannot be fulfilled on clusters.management.cattle.io "c-gdzsq": the object has been modified; please apply your changes to the latest version and try again
```

However, the call to update the cluster looks unnecessary because the [driverUpdate](https://github.com/rancher/rancher/blob/f2faf1a8aa1081301e1fe6c21637fe9b7044f188/pkg/controllers/management/clusterprovisioner/driver.go#L136-L139) function and its caller do not modify the cluster. Also, only the driverUpdate function returns the error, but driverRemove, driverUpdate, and driverCreate ignore the error.

### What was fixed, or what changes have occurred

Ignore the error if the update fails, like other functions mentioned above.
  
### Areas or cases that should be tested

etcd restore on an RKE1 cluster
  
### What areas could experience regressions?
  
etcd restore on an RKE1 cluster 

### Are the repro steps accurate/minimal?

yes

**Comment by jiaqiluo** [comment]
this issue can be validated on the latest v2.7-head

**Comment by thaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.7-head 5f98b0f
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: RKE1 node driver cluster

----------

### Testing: ###
Tested this issue with the following steps:
1. Deploy an RKE1 1.24 cluster with PSP enabled - set to Unrestricted
2. Take snapshot b1
3. Edit cluster upgrade k8s to 1.25. Disable PSP and save changes made
4. Cluster upgrades successfully
5. Restore to etcd snapshot b1 taken in step 2

**Result**
There is no longer an error when restoring the snapshot before the k8s upgrade and the restore is successful.
